### ISSUE TITLE
After updated my Mac, iPhone refuse show the screen in Vysor

### ISSUE THREAD

**Issue by hildenja** [koush/vysor.io#978]
macOS Monterey
Versio 12.0.1
With chip M1

Before the update, there wasn't any problems with Android/iOS mirroring on my Mac.
But now, the case is:
- Vysor recognize connected iPhone with it's idnumber as well
- but get stuck the view: "Connecting via Bluetooth (88-6b-******)"

I've tried every tricks related to BT connection issues:
- removed pairs from each devices; Mac/iPhone
- reconnect
- checked access privileges from Mac
- checked access privileges from iPhone

and now i am without any idea what to try next. Can you help me?

**Comment by mirrorme66** [comment]
Question: Can you mirror and control your IOS devices using vysor?

**Comment by hildenja** [comment]
No I can't, anymore. As I told:

"get stuck the view: "Connecting via Bluetooth (88-6b-******)"

**Comment by mirrorme66** [comment]
Interesting. So have you been able to  mirror and control ios devices using vysor in the past? Does one need a mac or will a windows server be ok?

**Comment by hildenja** [comment]
Have you read this issue report with thought?

"Before the update, there wasn't any problems with Android/iOS mirroring on my Mac."

**Comment by mirrorme66** [comment]
Yes i did and am asking some colleagues of mine if they can help. You only talking about mirroring....what about controlling?

**Comment by hildenja** [comment]
controlling the iPhone was possible as well. All features of Vysor worked well until I update the Mac OS.
I also tried reinstallation without success.

**Comment by clubtech** [comment]
This is happening to me as well when trying to mirror an iPhone on the latest MacOS Monterey. It was working fine before I updated the Mac.

**Comment by koush** [comment]
Let me double check.

**Comment by koush** [comment]
Seeing this as well on my M1 Max running Monterey. Looks like the some of the Mac bluetooth APIs are causing issues on Monterey. I'm running using the Vysor Dongle, so I never noticed. Let me see if I can reenable the mirroring only option as a stopgap... I'm hoping this is something I can fix.

**Comment by koush** [comment]
Can someone confirm if Vysor ever worked on Monterey at all with Mac Bluetooth? I tested briefly during the betas I think, but didn't keep up with it. Or did a recent Monterey update break it?

**Comment by koush** [comment]
There's now a bypass button so functionality is partially restored. Need further investigation on Monterey... gonna be a bit since I need gotta set up a dev environment on my M1X. Guess this is the push I needed.

![Koushik-iPhoneX](https://user-images.githubusercontent.com/73924/140562503-683aacbb-40a1-47b2-92c1-04dede18dd4f.jpg)
.

**Comment by koush** [comment]
> This is happening to me as well when trying to mirror an iPhone on the latest MacOS Monterey. It was working fine before I updated the Mac.

Intel or M1 Mac?

**Comment by koush** [comment]
I can't get my iPhone to pair with my M1, full stop. The connection will connect then disconnect a few seconds later. This is even outside of Vysor. Trying an OS update to 12.01 to see if that resolves it. I suspect this is an Apple bluetooth bug (possibly M1 only).

**Comment by clubtech** [comment]
> > This is happening to me as well when trying to mirror an iPhone on the latest MacOS Monterey. It was working fine before I updated the Mac.
> 
> Intel or M1 Mac?

This is on the new M1X Pro. There are also many other issues with BT the new OS that are not related to Vysor so I think apple has some new bugs in this release.

**Comment by koush** [comment]
> This is on the new M1X Pro. There are also many other issues with BT the new OS that are not related to Vysor so I think apple has some new bugs in this release.

If this is a major issue, you can buy one of the supported dongles. They're preferable really. Performance is way better.

**Comment by koush** [comment]
Will keep this open to track. I don't think there's a fix for Vysor here.

**Comment by hildenja** [comment]
Thanks for the clarification! I didn't immediately think of an OS bug. But I thought Vysoria wouldn’t just have been developed to be compatible. Indeed, I can no longer pair my iPhone with a Mac. However, Bluetooth is not completely out of the game. Unfortunately, now it seems that Apple products are not talking to each other, even if other devices work.

**Comment by wurikiji** [comment]
I got a sam problem here. I bought a dongle so I am looking forward to taking it and vysor working properly. 
@koush With a dongle, does vysor work great as usua?

**Comment by koush** [comment]
> I got a sam problem here. I bought a dongle so I am looking forward to taking it and vysor working properly.
> @koush With a dongle, does vysor work great as usual?

yes

**Comment by wurikiji** [comment]
> > I got a sam problem here. I bought a dongle so I am looking forward to taking it and vysor working properly.
> > @koush With a dongle, does vysor work great as usual?
> 
> yes

Great to hear that!

**Comment by bobfortin** [comment]
As mentioned previously in this thread MacOS Monterey seems to be having a few issues  with Bluetooth so I went out and purchased the MDK-Dongle. Following the instructions provided by Vysor (/koush/vysor.io/wiki/Vysor-Dongle#nrf52840-flashing-instructions) I am still unable to get a connection between my Mac or PC using the Dongle.  All I am getting is "Waiting on Bluetooth Device".   Screencast: https://www.screencast.com/t/PPEpg4t0p6vv

Please advise what I might be doing incorrectly.

**Comment by koush** [comment]
You need to reset the device settings or Vysor itself so it "forgets" your previous built in pairing that used the Mac bluetooth. That's my guess. Should work fine, since the screenshot indicates it was flashed properly.

**Comment by bobfortin** [comment]
> You need to reset the device settings or Vysor itself so it "forgets" your previous built in pairing that used the Mac bluetooth. That's my guess. Should work fine, since the screenshot indicates it was flashed properly.

Thanks for the quick response.  "Reset Device Settings" could you be a bit more specific about what you are referring to please.  I can Rest lots of items on my phone including resetting the entire device back to Factory Settings.  I did uninstall Vysor and reinstall it on my Mac - with no success.  I have tried three different iOS devices now to connect with no luck.

**Comment by koush** [comment]
I mean within vysor itself. Click reset vysor (will reset everything), or go into the vysor device settings and click defaults, save, close mirroring window, reopen.

**Comment by bobfortin** [comment]
> I mean within vysor itself. Click reset vysor (will reset everything), or go into the vysor device settings and click defaults, save, close mirroring window, reopen.

Thanks for the quick response:  I have:
- Reset Vysor per your instructions
- Set Settings to Default per you instructions

No love

Tried on the following devices and with the same result: _Waiting on bluetooth device..._
iPhone SE / iOS 13
iPhone 8 / iOS 14
iPhone 12 / iOS 15

I guess mirroring is my future... Very unfortunate.

**Comment by koush** [comment]
> I guess mirroring is my future... Very unfortunate.

Something's obviously wrong here. With the dongle, it should work. Doesn't matter what macOS version. Can you open a new bug and attach a bug report?

**Comment by koush** [comment]
You might need to simply unplug/replug the dongle to get it to kick in. Sometimes it's finicky after the first flash.

**Comment by koush** [comment]
Just did a sanity check, currently working on macOS bigsur/intel and macOS monterey/m1 with the NRF dongle.

**Comment by bobfortin** [comment]
> Just did a sanity check, currently working on macOS bigsur/intel and macOS monterey/m1 with the NRF dongle. iPhone 12 and 13 running 15.2.

Thank you, created a Bug Report from Vysor.  I am running Monterey 12.1 / Intel, I never had any issues connecting directly to Bluetooth when on Big Sur, Monterey has been a PITA and feels like an old Windows release (buggy). Please let me know if there is anything else I can provide.

**Comment by koush** [comment]
> > Just did a sanity check, currently working on macOS bigsur/intel and macOS monterey/m1 with the NRF dongle. iPhone 12 and 13 running 15.2.
> 
> Thank you, created a Bug Report from Vysor. I am running Monterey 12.1 / Intel, I never had any issues connecting directly to Bluetooth when on Big Sur, Monterey has been a PITA and feels like an old Windows release (buggy). Please let me know if there is anything else I can provide.

You will need to create a new issue here, and then attach the bug report in that issue.

**Comment by welkang** [comment]
Same issue: Monterey 12.1 and Intel.

**Comment by dprevost-LMI** [comment]
I have the same problem but no Dongle and not an M1 Mac. I updated on Sunday to Monterey and before that, it was working.

**Comment by chge** [comment]
Same here. Apparently, the AssistiveTouch hack has stopped working on Monterey for all the competitors as well (ApowerMirror, MirrorGo, Wormhole, you name it).

For those who desperate, restarting the Bluetooth daemon sometimes helps to set up the phone from scratch.
```shell
sudo pkill bluetoothd
```
(It restarts itself immediately.)

Not sure if it's related, but my nRF52840 Vysor Dongle is also unstable on Monterey. The Vysor app constantly loses it and falls back to built-in Bluetooth adapter until I plug it out and in again.

Thank you for the product anyway, it's probably the only one which offers a hardware option. Please keep going.

**Comment by kelvinsv** [comment]
I was having the same problem. Went to Advanced at the bottom of the vysor screen, selected Reset Vysor, and start working back again. When asked to pair the iPhone to my Mac, I chose to skip and only mirror. 

Working fine now! Hope that helps.

**Comment by clubtech** [comment]
Anyone with 12.3 can confirm if this has been resolved or not in this release?

**Comment by bobfortin** [comment]
Upgraded to 12.3 and tried connecting iPhone to Bluetooth and the issue still exists as previously noted.

**Comment by koush** [comment]
This is a Monterey issue, and may never be fixed. Apple prevents iPhone from connecting to Mac. Locking this.

https://github.com/koush/vysor.io/issues/978#issuecomment-962220191
